src/services/refactors/convertImport.ts

```diff
@@ -0,0 +1,113 @@
+
+        // We may need to change `mod.x` to `_x` to avoid a name conflict.
+        const exportNameToImportName = createMap<string>();
+        let usedAsNamespace = false;
+
+        forEachFreeIdentifier(sourceFile, id => {
+            if (id === toConvert.name || checker.getSymbolAtLocation(id) !== namespaceSymbol) return;
+
+            if (!isPropertyAccessExpression(id.parent)) {
+                usedAsNamespace = true;
+            }
+            else {
+                const parent = cast(id.parent, isPropertyAccessExpression);
+                const exportName = parent.name.text;
+                let name = exportNameToImportName.get(exportName);
+                if (name === undefined) {
+                    exportNameToImportName.set(exportName, name = generateName(exportName, usedIdentifiers));
+                }
+                Debug.assert(parent.expression === id);
+                changes.replaceNode(sourceFile, parent, createIdentifier(name));
+            }
+        });
+
+        const elements: ImportSpecifier[] = [];
+        exportNameToImportName.forEach((name, propertyName) => {
+            elements.push(createImportSpecifier(name === propertyName ? undefined : createIdentifier(propertyName), createIdentifier(name)));
+        });
+        const namedImports = createNamedImports(elements);
+
+        if (usedAsNamespace) {
+            changes.insertNodeAfter(sourceFile, toConvert.parent.parent,
+                createImportDeclaration(/*decorators*/ undefined, /*modifiers*/ undefined, createImportClause(/*name*/ undefined, namedImports), toConvert.parent.parent.moduleSpecifier));
+        }
+        else {
+            changes.replaceNode(sourceFile, toConvert, namedImports);
+        }
+    }
+
+    function doChangeNamedToNamespace(sourceFile: SourceFile, checker: TypeChecker, changes: textChanges.ChangeTracker, toConvert: NamedImports, usedIdentifiers: ReadonlyMap<true>): void {
+        const { moduleSpecifier } = toConvert.parent.parent;
+        // We know the user is using at least ScriptTarget.ES6, and moduleSpecifierToValidIdentifier only cares if we're using ES5+, so just set ScriptTarget.ESNext
+        const namespaceImportName = generateName(moduleSpecifier && isStringLiteral(moduleSpecifier) ? codefix.moduleSpecifierToValidIdentifier(moduleSpecifier.text, ScriptTarget.ESNext) : "module", usedIdentifiers);
+
+        changes.replaceNode(sourceFile, toConvert, createNamespaceImport(createIdentifier(namespaceImportName)));
+
+        const nameToPropertyName = createMap<string>();
+        for (const element of toConvert.elements) {
+            if (element.propertyName) {
+                nameToPropertyName.set(element.name.text, element.propertyName.text);
+            }
+        }
+
+        const importedSymbols = toConvert.elements.map(e => checker.getSymbolAtLocation(e.name)!);
+
+        forEachFreeIdentifier(sourceFile, id => {

```

Comment:
why not just find all references on the ImportSpecifier and change these instead of looking at all identifiers..

Reply:
also we need to handle cases where the propertyAccess expression will not work.. e.g. shorthandproperties, `var x = {a}`, changing it to `var x = { m.a }` will not work, you need to change it to `var x = { a: m.a }`. similarly the export declarations need to be handled separately.. so `export {a}` will have to be rewritten either as ` export {a} from "mod"` or `import a = m.a; export {a};`